docs/Autoscaling.md

```diff
@@ -0,0 +1,150 @@
+## What is
+Autoscaling is an optional feature in which the user can choose and parametrize different autoscaling policies that maestro
+will use to automatically scale the number of rooms in the scheduler.
+
+Maestro has an internal process that periodically keeps checking if it needs to create or delete game rooms for the given scheduler,
+if autoscaling is not configured or enabled, it will always try to maintain the current number of rooms equal to **roomsReplicas** scheduler property.
+If autoscaling is configured **and** enabled, it will use the configured autoscaling policy to decide if it needs to **scale up** (create more rooms), 
+**scale down** (delete rooms) or **do nothing**.
+
+```mermaid
+  flowchart TD
+    finish((End))
+    add_rooms_operation(Enqueue add rooms operation)
+    remove_rooms_operation(Enqueue remove rooms)
+    use_rooms_replicas(Use rooms replicas to calculate the desired number of rooms)
+    autoscaling_enabled{Autoscaling configured and enabled?}
+    decide_operation{Compare current number of rooms with the desired amount.}
+    use_autoscaling[Use Autoscaling policy to calculate the desired number of rooms corced in min max range]
+    autoscaling_enabled -- No --> use_rooms_replicas;
+    autoscaling_enabled -- Yes --> use_autoscaling;
+    use_autoscaling --> decide_operation;
+    use_rooms_replicas --> decide_operation;
+    decide_operation --  desired > actual --> add_rooms_operation --> finish;
+    decide_operation --  desired == actual --> finish;
+    decide_operation --  desired < actual --> remove_rooms_operation --> finish;
+```
+
+Currently, the sync interval is configured by environment variable `MAESTRO_WORKERS_OPERATIONEXECUTION_HEALTHCONTROLLERINTERVAL`.
+
+> By default, the scheduler does not have autoscaling configured.
+
+### How to configure and enable autoscaling
+To get autoscaling working in your scheduler, firstly you need to configure an autoscaling policy and enable it, this autoscaling
+configuration resides in the root of the scheduler structure itself.
+
+Yaml version
+```yaml
+name: String
+game: String
+...
+autoscaling:
+  enabled: true
+  min: 1
+  max: 10
+  policy:
+    type: roomOccupancy
+    parameters:
+      ...
+      // Will vary according to the policy type.
+```
+
+Json version
+```json
+{
+  "name": "test",
+  "game": "multiplayer",
+  ...
+  "autoscaling": {
+    "enabled": true,
+    "min": 10,
+    "max": 300,
+    "policy": {
+      "type": "roomOccupancy",
+      "parameters": {
+        ...
+        // Will vary according to the policy type.
+      }
+    }
+  }
+}
+```
+
+- **enabled** [boolean]: A value that can be true or false, indicating if the autoscaling feature is enabled/disabled for the given scheduler. Default: false.
+- **min** [integer]: Minimum number of rooms the scheduler should have, it must be greater than zero.
```

Comment:
```suggestion
- **min** [integer]: Minimum number of rooms the scheduler should have, it must be greater than zero. For zero value, disable autoscaling and set "roomsReplicas" to 0.
```

Reply:
Trust me, this information is very relevant for people rolling out the docs trying to understand how to have 0 replicas